pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,210 @@
+			// the dependency graph for each of them (if the user has specified the
+			// --all flag)
+			repoList, err := c.ImageRepositories(namespace).List(labels.Everything(), labels.Everything())
+			checkErr(err)
+
+			repos := make([]string, 0)
+			if all {
+				for _, repo := range repoList.Items {
+					repos = append(repos, repo.Name)
+				}
+			} else {
+				repos = append(repos, args[0])
+				// Validate the specified image repository
+				if !isValid(repos[0], repoList.Items) {
+					checkErr(fmt.Errorf("invalid image repository: %s", repos[0]))
+				}
+			}
+
+			// Get all build configurations
+			buildConfigList, err := c.BuildConfigs(namespace).List(labels.Everything(), labels.Everything())
+			checkErr(err)
+
+			output := cmdutil.GetFlagString(cmd, "output")
+			for _, repo := range repos {
+				root := findRepoDeps(repo, namespace, buildConfigList.Items)
+
+				switch output {
+				case "json":
+					jsonDump, err := json.MarshalIndent(root, "", "\t")
+					checkErr(err)
+					fmt.Println(string(jsonDump))
+				case "dot":
+					g := dot.NewGraph()
+					g.SetName("G")
+					// Directed graph since we illustrate dependencies
+					g.SetDir(true)
+					g.AddAttr("G", "namespace", namespace)
+					fmt.Println(dotDump(root, g))
+				case "ast":
+					fmt.Println(root)
+				default:
+					checkErr(fmt.Errorf("wrong output format specified: %s", output))
+				}
+			}
+		},
+	}
+
+	cmd.Flags().Bool("all", false, "Build dependency trees for all image repositories")
+	cmd.Flags().StringP("output", "o", "json", "Output format of dependency tree(s)")
+	return cmd
+}
+
+// isValid checks if the given image repository is valid
+func isValid(name string, repoList []imageapi.ImageRepository) bool {
+	for _, repo := range repoList {
+		if name == repo.Name {
+			if errs := val.ValidateImageRepository(&repo); len(errs) == 0 {
+				return true
+			}
+			break
+		}
+	}
+
+	return false
+}
+
+// findRepoDeps accepts an image repository name and a list of build
+// configurations and returns the dependency tree of the specified
+// image repository
+func findRepoDeps(from, namespace string, buildConfigList []buildapi.BuildConfig) *ImageRepo {
+	root := &ImageRepo{
+		Name:      from,
+		Namespace: namespace,
+	}
+
+	// Search all build configurations in order to find the image
+	// repositories depending on the specified image repository
+	var name, tag string
+	var err error
+	for _, cfg := range buildConfigList {
+		for _, trigger := range cfg.Triggers {
+			// If an ImageChangeTrigger has been specified, use its From field
+			// to compare against the specified image repository
+			if trigger.ImageChange != nil && from == trigger.ImageChange.From.Name {
+				// Either DockerImageReference or To & Tag will be used as output
+				if cfg.Parameters.Output.To.Name == "" {
+					_, _, name, tag, err = imageapi.SplitDockerPullSpec(cfg.Parameters.Output.DockerImageReference)
+					checkErr(err)
+				} else {
+					name = cfg.Parameters.Output.To.Name
```

Comment:
I believe it's valid for Output.To to specify a namespace (Output.To.Namespace) so this should take that into account.  (if not specified it defaults to this namespace)